src/metabase/driver/sql_jdbc/connection.clj

```diff
@@ -87,7 +87,9 @@
   (let [details-with-tunnel (ssh/include-ssh-tunnel details) ;; If the tunnel is disabled this returned unchanged
         spec                (connection-details->spec driver details-with-tunnel)
         properties          (data-warehouse-connection-pool-properties driver)]
-    (connection-pool/connection-pool-spec spec properties)))
+    (connection-pool/connection-pool-spec spec (merge properties
+                                                      (when (ssh/use-ssh-tunnel? details)
+                                                        {"idleConnectionTestPeriod" 120})))))
```

Comment:
can this be done the ssh code itself? somewhere like here: https://github.com/metabase/metabase/blob/master/src/metabase/util/ssh.clj#L43-L59

This is a good fix - because the tunnels do timeout when there's no "heartbeat" or traffic, but with it here, MongoDB and other databases that don't derive from sql_jdbc won't benefit from it.

Reply:
I've made the change. As it stands it is hardcoded to 180 seconds. Will make a new ticket so this is user settable when configuring. But might be worth it to accept the current implementation.